daemon/algod/server.go

```diff
@@ -268,6 +269,19 @@ func makeListener(addr string) (net.Listener, error) {
 	return net.Listen("tcp", addr)
 }
 
+// helper to get port from an address
+func getPortFromAddress(addr string, logger logging.Logger) string {
```

Comment:
consider returning `fmt.Errorf("error parsing address: %w", err) and checking in the caller - functions without side I guess are preferable.

Reply:
Errors are now logged in the caller, and with error returned by this function